site/testsuites/testsuite-javascript/__tests__/tbdex/pfi/quotes.js

```diff
@@ -0,0 +1,285 @@
+
+
+  async write({ message }) {
+      await dataProvider.insert('exchange', {
+              exchangeid: message.exchangeId,
+              messagekind: message.kind,
+              messageid: message.id,
+              subject: message.subject,
+              message: JSON.stringify(message)
+          });
+  }  
+
+};
+
+const offeringsApiProvider = {
+  getOffering: (opts) => {
+      dataProvider.get('offering', opts.id).then(([result]) => {
+          if (!result) {
+              return undefined
+          }
+          return Offering.factory(result.offering)
+      });
+  },
+
+  getOfferings: (opts) => {
+    dataProvider.query('offering', "*").then((results) => {
+        const offerings = []
+
+        for (let result of results) {
+            const offering = Offering.factory(result.offering)
+            offerings.push(offering)
+        }
+
+        return offerings
+    });
+  },
+
+
+  async create(offering) {
+      await dataProvider.insert('offering', {
+          offeringid: offering.id,
+          payoutcurrency: offering.payoutCurrency.currencyCode,
+          payincurrency: offering.payinCurrency.currencyCode,
+          offering: JSON.stringify(offering)
+      });
+  }
+  
+};  
+
+//---------------------------------------------------------------------------//
+//---------------------------------------------------------------------------//
+
+async function createQuoteFromRfq(message) {
+    // :snippet-start: pfiQuotesWriteJs
+    // Write the message to your exchanges database
+    await dataProvider.insert('exchange', {
+        exchangeid: message.exchangeId,
+        messagekind: message.kind,
+        messageid: message.id,
+        subject: message.subject,
+        message: JSON.stringify(message)
+    });
+
+    //highlight-start
+    const offering = await offeringsApiProvider.getOffering(message.offeringId)
+    //highlight-end
+    // :snippet-end:
+
+    // :snippet-start: pfiQuotesProcessJs
+    const rfqOptions = {
+        data: message.data(), 
+        metadata: message.metadata()
+    };
+
+    const rfq = Rfq.create(rfqOptions)
+
+    try {
+        await rfq.verifyOfferingRequirements(offering)
+    } catch(e) {
+        console.log(`Failed to verify offering requirements: ${e.rfq}`)
+    }
+    // :snippet-end:
+
+    // :snippet-start: pfiQuotesSendJs
+    const quote = Quote.create(
+        {
+          metadata: {
+            from: config.did.id,
+            to: message.from,
+            exchangeId: message.exchangeId
+          },
+          data: {
+            expiresAt: new Date(2024, 4, 1).toISOString(),
+            payin: {
+              currencyCode: 'BTC',
+              amountSubunits: '1000'
+            },
+            payout: {
+              currencyCode: 'KES',
+              amountSubunits: '123456789'
+            }
+          }
+        }
+    );
+    // :snippet-end:
+
+    // :snippet-start: pfiQuotesSignJs
+    await quote.sign(config.did.privateKey, config.did.kid)
```

Comment:
`sign` takes a did now. https://tbd54566975.github.io/tbdex-kt/protocol/tbdex.sdk.protocol.models/-message/sign.html

```suggestion
    await quote.sign(pfiDid)
```